app/controllers/api/v1/shifts_controller.rb

```diff
@@ -0,0 +1,66 @@
+module Api
+  module V1
+    class ShiftsController < ApplicationController
+      # skip_before_action :verify_authenticity_token
+      before_action :set_shift, only: %i[ show update destroy ]
+
+      def index
+        shifts = Shift.all
+        render json: ShiftSerializer.new(shifts).serialized_json, status: :ok
+      end
+
+      def show
+        if @shift
+          render json: ShiftSerializer.new(@shift).serialized_json, status: :ok
+        else
+          render json: { error: "Unable to update shift" }, status: :unprocessable_entity
+        end
+      end
+
+      def create
+        user = User.find_by(id: shift_params[:user_id])
+        schedule = Schedule.find_by(id: shift_params[:schedule_id])
+        shift = Shift.new(
+          user: user,
+          schedule: schedule,
+          shift_start: shift_params[:shift_start],
+          shift_end: shift_params[:shift_end],
+        )
+
+        if shift.save
+          render json: { message: "Shift for #{shift.user.employee_name} created: #{shift.format_shift(shift.shift_start)} - #{shift.format_shift(shift.shift_end)}" }, status: :created
+        else
+          render json: { error: "Unable to create shift" }, status: :unprocessable_entity
```

Comment:
To drive some sort of action on the part of the user, you could pass the reason(s) for the inability to create shift here. When `shift.save` fails, it will populate the errors in `shift.errors`.  If you pass that along, the frontend could format and render them cleanly.